validator/client/beacon-api/activation_test.go

```diff
@@ -0,0 +1,197 @@
+//go:build use_beacon_api
```

Comment:
The case where a non-200 status is returned doesn't seem to be here. it would be good to test that scenario with both the valid error json case, and the badly formatted error json case.

Reply:
All cases (200, 400, 404, 500, others) are now checked.
About timeout, I don't want the test to last 12+ secs.

==> I Check `<- time.After(waitDuration` is blocking when `Recv` is called two times in a raw, and I check the computation of `waitDuration` (moved in a new dedicated function) is done correctly.

(Fixed a bug on this computation)